src/dockerfile.ml

```diff
@@ -243,6 +295,65 @@ let parser_directive pd : t = [ `ParserDirective pd ]
 let heredoc ?(strip = false) ?(word = "EOF") ?(delimiter = word) fmt =
   ksprintf (fun here_document -> { here_document; strip; word; delimiter }) fmt
 
+let mount_bind ~target ?source ?from ?readwrite () =
+  let m = { target; source; from; readwrite } in
+  { typ = `Bind m }
+
+let mount_cache ?id ~target ?readonly ?sharing ?from ?source ?mode ?uid ?gid ()
+    =
+  let m = { id; target; readonly; sharing; from; source; mode; uid; gid } in
+  { typ = `Cache m }
+
+let mount_tmpfs ~target ~size =
+  let m = { target; size } in
+  { typ = `Tmpfs m }
+
+let mount_secret ?id ?target ?required ?mode ?uid ?gid () =
+  let m = { id; target; required; mode; uid; gid } in
+  { typ = `Secret m }
+
+let mount_ssh ?id ?target ?required ?mode ?uid ?gid () =
+  let m = { id; target; required; mode; uid; gid } in
+  { typ = `Ssh m }
+
+let mount_args { typ } =
+  match typ with
+  | `Bind { target; source; from; readwrite } ->
+      String.concat ","
+        ([ "--mount=type=bind" ]
+        @ [ sprintf "target=%s" target ]
+        @ optional "source" source @ optional "from" from
+        @ optional_bool "readwrite" readwrite)
+  | `Cache { id; target; readonly; sharing; from; source; mode; uid; gid } ->
+      String.concat ","
+        ([ "--mount=type=cache" ] @ optional "id" id
+        @ [ sprintf "target=%s" target ]
+        @ optional_bool "readonly" readonly
+        @ (match sharing with
+          | None -> []
+          | Some `Shared -> [ "sharing=shared" ]
+          | Some `Private -> [ "sharing=private" ]
+          | Some `Locked -> [ "sharing=locked" ])
+        @ optional "from" from @ optional "source" source
+        @ optional_int_octal "mode" mode
+        @ optional_int "uid" uid @ optional_int "gid" gid)
+  | `Tmpfs { target; size } ->
+      sprintf "--mount=type=tmpfs,target=%s,size=%d" target size
```

Comment:
size is only supported by Docker. I don't think Podman supports size, just target, so perhaps size could be made optional?

Reply:
Done, thanks!